include/deal.II/grid/grid_generator.h

```diff
@@ -1579,13 +1579,9 @@ namespace GridGenerator
    * @return A map that for each cell of the surface mesh (key) returns an
    * iterator to the corresponding face of a cell of the volume mesh (value).
    * The keys include both active and non-active cells of the surface mesh.
-   * For dim=2 (i.e., where volume cells are quadrilaterals and surface cells
-   * are lines), the order of vertices of surface cells and the corresponding
-   * volume faces match. For dim=3 (i.e., where volume cells are hexahedra and
-   * surface cells are quadrilaterals), the order of vertices may not match in
-   * order to ensure that each surface cell has a right-handed coordinate
-   * system when viewed from one of the two sides of the surface connecting
-   * the cells of the surface mesh.
+   * The order of vertices of surface cells and the corresponding
+   * volume faces may not match in order to ensure that each surface cell is
+   * associated with an outward facing normal.
```

Comment:
Agreed. But as I have written in my comment above, translating vertices and quadrature points to match quantities on boundary of the volume mesh and on the surface mesh is cumbersome. I am wondering, whether we could do better by setting direction_flag appropriately in extract_boundary_mesh instead of swapping vertices? In particular, I believe that the concept in deal.ii that cells, faces and lines are stored as separate quantities is very reasonable. This way also all codim>0 meshes are in some sense already include as a "subset" in a codim=0 mesh. By swapping vertices, this nice property is  destroyed more or less ...

Reply:
I suspect you can always come up with cases where you need some kind of translation between vertices. The problem is that we want to impose certain properties in the meshes we have and asking for too many properties to be true at the same time may make the set of meshes we can represent zero. I'm not saying that what you suggest is not possible, but I would not be surprised if you can't always do it that way.

Reply:
I am not sure whether I really understand what you say. The only case I can think of, where the approach I am suggesting won't work is if the there are lines with non standard orientation on the boundary of a volume mesh. But this doesn't work with the current implementation of extract_boundary_mesh either (with the current implementation, it could probably made work by doing reordering of the grid if lines with non standard orientation are present). With the approach I am suggesting, one would have to allow line_direction to be false for a 3d codim 1 mesh - I suspect that it would indeed be difficult to make this work. Is this what you mean? Or is it maybe not possible to arbitrarily set the direction_flag for a 3d codim 1 mesh (i.e., are there constraints on how the direction_flag of neighboring cells is set)?

Reply:
You catch me flatfooted. There are probably only 3 or so people who really understood how these direction flags work. I am one of those, but I don't remember the details of it all any more. So if you think it can work, then I would say try it out!

I am an incrementalist. So one of the possible ways forward would be to finish the patch in its current form (assuming there is anything to finish -- I haven't looked), get it merged, and then do as you suggest with the direction flags. This way, there is a known and working state in the repository that's already an improvement over what's in 9.0. Then we can see if your idea pans out: if it does, great, which we can check with a few additional tests; if it doesn't, then at least we can stick with the state of the current PR which by then is in the repository.

Reply:
Good, I think the approach you propose is reasonable. Let's do it like this. For now it remains only to fix some tests which are broken by the patch.

Reply:
OK. How do we go about this then? If you are confident that the patch is correct -- which I think is credible -- can you just update the test outputs?

Reply:
Most of the tests should fail either because normal vectors are output or the mesh is written to a file. It's fairly easy to check that this is indeed the case and this is what I'll definitely do plus a few plausibility checks and visual inspection of the meshes produced. I have already fixed 20 or so broken tests this way and I'll do it the same with the remaining 12. Doing real in depth checking of every single test would take ages ...

Reply:
I think it's ok if a test passes the smell check. If the changes in 20 tests are all to be expected, the likelihood that the remaining 12 just follow the same pattern is quite high and I don't see why you need to look at every single one in great detail -- just update the output files.